examples/gen-ai-hello-world/local-start.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+PYTHON_VERSION="3.11"
+POETRY_VERSION="1.8.1"
+
+log() {
+    # Logs a message with a timestamp to both the console and a log file.
+    echo "$(date '+%Y-%m-%d %H:%M:%S') - $1"
+}
+
+exit_application() {
+    # Handles script termination, logging a shutdown message and exiting with code 1.
+    echo -e "\033[31m"
+    log "Exiting..."
+    echo -e "\033[0m"
+
+    exit 1
+}
+
+handle_error() {
+    # Logs an error message and shuts down the application.
+    echo -e "\033[31m"
+    log "An error occurred: $1"
+    echo -e "\033[0m"
+    exit_application
+}
+
+trap handle_error ERR
+
+check_command_version() {
+    # Checks if a command is installed and meets the required version.
+    # Arguments:
+    #   $1 - Command name
+    #   $2 - Required version
+    #   $3 - Version flag (e.g., --version)
+    local cmd=$1
+    local required_version=$2
+    local version_flag=$3
+    local current_version
+
+    if ! command -v "$cmd" >/dev/null 2>&1; then
+        handle_error "$cmd is not installed. Please install $cmd version $required_version or above."
+    else
+        current_version=$("$cmd" "$version_flag" 2>&1 | grep -Eo '[0-9]+(\.[0-9]+)+')
+        if [[ -z "$current_version" ]]; then
+            handle_error "Could not determine $cmd version. Please ensure it is installed correctly."
+        fi
+        if [[ "$(printf '%s\n' "$required_version" "$current_version" | sort -V | head -n1)" != "$required_version" ]]; then
+            handle_error "$cmd version $required_version or above is required. You have version $current_version."
+        fi
+    fi
+
+    log "$cmd is installed and meets the required version ($required_version). Current version: $current_version."
+}
+
+check_requirements() {
```

Comment:
Let's also check for `gcloud` pre-requisite.

Reply:
updated ya [5358986](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/5358986f97b9fee950b140d31788e934846b9137)
I also slightly modify the regex to grep the version